Microsoft.HBase.Client/Requester/VNetWebRequester.cs

```diff
@@ -49,40 +49,33 @@ public WebRequesterBasic(string contentType = "application/x-protobuf")
         /// <param name="endpoint">The endpoint.</param>
         /// <param name="method">The method.</param>
         /// <param name="input">The input.</param>
+        /// <param name="options">request options</param>
         /// <returns></returns>
-        public HttpWebResponse IssueWebRequest(string endpoint, string method = "GET", Stream input = null)
+        public HttpWebResponse IssueWebRequest(string endpoint, string method, Stream input, RequestOptions options)
         {
-            var response =  IssueWebRequestAsync(endpoint, method, input).Result;
-            
-            return response;
+            return IssueWebRequestAsync(endpoint, method, input, options).Result;
         }
-        
+
         /// <summary>
         /// Issues the web request asynchronous.
         /// </summary>
         /// <param name="endpoint">The endpoint.</param>
         /// <param name="method">The method.</param>
         /// <param name="input">The input.</param>
-        /// <param name="alternativeEndpointBase">The alternative endpoint base.</param>
+        /// <param name="options">request options</param>
         /// <returns></returns>
-        public async Task<HttpWebResponse> IssueWebRequestAsync(string endpoint, string method = "GET", Stream input = null, string alternativeEndpointBase = null)
+        public async Task<HttpWebResponse> IssueWebRequestAsync(string endpoint, string method, Stream input, RequestOptions options)
         {
             Trace.CorrelationManager.ActivityId = Guid.NewGuid();
-
-            Task<HttpWebResponse> result = new Task<HttpWebResponse>(() => { return null; });
-
-            if (alternativeEndpointBase == null)
-            {
-                // alternativeEndpointBase needed in VNET mode. Abort!
-
-                return await result as HttpWebResponse;
-            }
-            
-            var target = new Uri(new Uri(alternativeEndpointBase), endpoint);
+            var target = new Uri(_balancer.GetEndpoint(), options.AlternativeEndpoint + endpoint);
```

Comment:
this guy needs to take care of the lifecycle around the load balancing mechanism as well. Follow-up ticket too.